File: code/snippets/swap-chars/swap.rb
# strategy: chain method
# calls with placeholder
def swap_chars(str, char1, char2, temp = '¢')
  str.gsub(char1, temp)
    .gsub(char2, char1)
    .gsub(temp, char2)
end

# prints @rolling_codes
puts swap_chars('@lorring_codes', 'l', 'r')
# prints 2520 im binary
puts swap_chars('1111011000100111', '1', '0')

File: code/snippets/swap-chars/swap.ts
// strategy: use regex with
// js lambda functions 
function swapChars(str: string, a: string, b: string) {
  return str.replace(
    new RegExp(`${a}|${b}`, 'g'), 
    ($0) => $0 === a ? b : a);
}

// prints @rolling_codes
console.log(
  swapChars(swapChars(swapChars(
    '@gillons_cider', 
      's', 'g'), 
        'r', 's'), 
          'o', 'i'));
// prints 69 in binary
console.log(
  swapChars('1111111110111010', '0', '1'));